examples/src/bin/texture.rs

```diff
@@ -264,7 +264,8 @@ fn main() {
         let image = image::load_from_memory(include_bytes!("../../assets/rust.png"))
             .unwrap()
             .to_rgba8();
-        let image_dimensions = image.dimensions();
+        let (width, height) = image.dimensions();
+        let image_dimensions = vk::Extent2D { width, height };
```

Comment:
could add and use:
```
impl From<(u32, u32)> for vk::Extent2D {
    fn from(value: (u32, u32)) -> Self {
        Self {
            width: value.0,
            height: value.1,
        }
    }
}
````

Reply:
I like the idea of conversions to/from tuples because of how it'd make interop simpler, though both @MarijnS95 and @Ralith have told me that having a small minimally-opinionated API is a goal, so I'm a little reticent to solve more than my biggest pain points until additional people express enthusiasm for tuple conversions.

Reply:
If we were to pursue this (and I don't find it super compelling) then we should do it as a separate PR so as not to bog down this one. For most purposes I favor nominal over structural typing, for the usual reasons.